packages/platform-node/examples/api.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  Api,
+  ApiBuilder,
+  ApiClient,
+  ApiEndpoint,
+  ApiGroup,
+  ApiSchema,
+  ApiSecurity,
+  ApiSwagger,
+  HttpClient,
+  HttpMiddleware,
+  HttpServer,
+  OpenApi
+} from "@effect/platform"
+import { NodeHttpServer, NodeRuntime } from "@effect/platform-node"
+import { Schema } from "@effect/schema"
+import { Context, Effect, Layer, Redacted } from "effect"
+import { createServer } from "node:http"
+
+class User extends Schema.Class<User>("User")({
+  id: Schema.Number,
+  name: Schema.String
+}) {}
+
+class CurrentUser extends Context.Tag("CurrentUser")<CurrentUser, User>() {}
+
+class Unauthorized extends Schema.TaggedError<Unauthorized>()("Unauthorized", {
+  message: Schema.String
+}, ApiSchema.annotations({ status: 401 })) {}
+
+const security = ApiSecurity.bearer
+
+const securityMiddleware = ApiBuilder.middlewareSecurity(
+  security,
+  CurrentUser,
+  (token) => Effect.succeed(new User({ id: 1000, name: `Authenticated with ${Redacted.value(token)}` }))
+)
+
+const users = ApiGroup.make("users").pipe(
+  ApiGroup.add(
+    ApiEndpoint.get("findById", "/:id").pipe(
+      ApiEndpoint.setPath(Schema.Struct({
+        id: Schema.NumberFromString
+      })),
+      ApiEndpoint.setSuccess(User),
+      ApiEndpoint.addError(Schema.String.pipe(
+        ApiSchema.asEmpty({ status: 413, decode: () => "boom" })
+      ))
+    )
+  ),
+  ApiGroup.add(
+    ApiEndpoint.post("create", "/").pipe(
+      ApiEndpoint.setPayload(ApiSchema.Multipart(Schema.Struct({
```

Comment:
Are all this supposed to set validation schemas? If that is the case (and I'm being a bit nitpicking here) reflecting that in the name will be easier to understand and more consistent with the rest of the methods of the platform package (eg setCookiesSchema or setHeadersSchema)

Reply:
All Api* modules are schema only, and the `ApiBuilder` module is for the implementation side.

The Api naming might change to make it more obvious that everything is a schema.